typhon/positioner.py

```diff
@@ -0,0 +1,190 @@
+            reload_widget_stylesheet(self, cascade=True)
+            raise_to_operator(exc)
+
+    @Slot()
+    def positive_tweak(self):
+        """Tweak positive by the amount listed in ``ui.tweak_value``"""
+        setpoint = self._get_position() + float(self.tweak_value.text())
+        self.ui.set_value.setText(str(setpoint))
+        self.set()
+
+    @Slot()
+    def negative_tweak(self):
+        """Tweak negative by the amount listed in ``ui.tweak_value``"""
+        setpoint = self._get_position() - float(self.tweak_value.text())
+        self.ui.set_value.setText(str(setpoint))
+        self.set()
+
+    @Slot()
+    def stop(self):
+        """Stop device"""
+        for device in self.devices:
+            device.stop()
+
+    def _get_position(self):
+        if not self._readback:
+            raise Exception("No Device configured for widget!")
+        return self._readback.get()
+
+    def add_device(self, device):
+        """Add a device to the widget"""
+        # Add device to cache
+        self.devices.clear()  # only one device allowed
+        super().add_device(device)
+        # Limit switches
+        for limit_switch in ('low_limit_switch',
+                             'high_limit_switch'):
+            # If our device has a limit switch attach it
+            if getattr(device, limit_switch, False):
+                widget = getattr(self.ui, limit_switch)
+                limit = getattr(device, limit_switch)
+                limit_chan = channel_from_signal(limit)
+                register_signal(limit)
+                widget.channel = limit_chan
+            # Otherwise, hide it the widget
+            else:
+                getattr(self.ui, limit_switch).hide()
+        # User Readback
+        if hasattr(device, 'user_readback'):
+            self._readback = device.user_readback
+        else:
+            # Let us assume it is the first hint
+            self._readback = grab_kind(device, 'hinted')[0][1]
+        register_signal(self._readback)
+        self.ui.user_readback.channel = channel_from_signal(self._readback)
+        # Limit values
+        # Look for limit signals first
+        if isinstance(device, Device):
+            limit_signals = ('low_limit' in device.component_names
+                             and 'high_limit' in device.component_names)
+            # Use raw signals to keep widget updated
+            if limit_signals:
+                register_signal(device.low_limit)
+                low_lim_chan = channel_from_signal(device.low_limit)
+                self.ui.low_limit.channel = low_lim_chan
+                register_signal(device.high_limit)
+                high_lim_chan = channel_from_signal(device.high_limit)
+                self.ui.high_limit.channel = high_lim_chan
+                return
+        # Check that we have limits at all, or if they are implemented
+        if hasattr(device, 'limits') and device.limits[0] != device.limits[1]:
```

Comment:
```suggestion
        if hasattr(device, 'limits') and device.limits[0] < device.limits[1]:
```

Reply:
Will this work with `(None, None)`?

Reply:
There's an assumption in `ophyd`'s positioner classes that the limits are never `(None, None)`, in fact they default to `(0, 0)`